site/publish.sh

```diff
@@ -47,15 +47,24 @@ function parallelfunction() {
         ;;
 
     azsync*)
+        # Load the env variables corresponding to use get-fileshare-signed-url.sh for the Azure File Share to sync, extracted from the end of the task name
+        envToLoad=".env-${1#azsync-}"
+        # shellcheck source=/dev/null
+        source "${envToLoad}"
+
+        # Required variables that should now be set from the .env file
+        : "${STORAGE_FILESHARE?}" "${JENKINS_INFRA_FILESHARE_CLIENT_ID?}" "${JENKINS_INFRA_FILESHARE_CLIENT_SECRET?}" "${FILESHARE_SYNC_SOURCE?}"
+
         # Script stored in /usr/local/bin used to generate a signed file share URL with a short-lived SAS token
         # Source: https://github.com/jenkins-infra/pipeline-library/blob/master/resources/get-fileshare-signed-url.sh
         fileShareUrl=$(get-fileshare-signed-url.sh)
```

Comment:
https://github.com/jenkins-infra/pipeline-library/pull/854 to ensure we can call 2 concurent `azsync`  tasks